README.md

```diff
@@ -83,9 +83,29 @@ juju config blackbox-exporter-k8s \
   probes_file='@path/to/probes.yml'
 ```
 
-Note that the `relabel_configs` of each scrape job doesn't need to be specified, and will be 
+Note that the `relabel_configs` of each scrape job doesn't need to be specified, and will be
 overridden by the charm with the needed labels and the correct Blackbox Exporter url.
 
+#### Dynamic Configuration
+
+The list of probes and the list of modules for probing can also be changed dynamically from other charms.
+The charm offers a relation to allow charms to forward custom probe spec to Blackbox Exporter. Those are exported over the probes relation using the blackbox_exporter_probes interface:
+
+```shell
+requires:
+  probes:
+    interface: blackbox_exporter_probes
+```
+
+The cusrom probes provided via relation data are merged with the probes defined in a configuration file, same with the modules which are integrated in the blackbox-config file.
```

Comment:
We should mention this also in the description of the [config options](https://github.com/canonical/blackbox-exporter-k8s-operator/blob/main/config.yaml).

Reply:
added in 8eaaac8170f5fdc10c1d2483b154b262fa145ead